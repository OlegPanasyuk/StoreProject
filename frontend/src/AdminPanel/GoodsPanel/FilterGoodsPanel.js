import React, { Component } from 'react';
import { Form } from 'react-bootstrap';

export class FilterGoodsPanel extends Component {
    constructor(props) {
        super(props);
        this.state = {
            arrOfCatalogue: []
        };
    }

    UNSAFE_componentWillMount() {
        const  self = this;
        if (fetch) {
            let myInit = {
                method: 'GET',
            };
            fetch(`${process.env.REACT_APP_API_HOST}:${process.env.REACT_APP_API_PORT}/catalogue`, myInit)
                .then(res => {
                    return res.json();
                })
                .then(data=>{
                    self.setState({
                        arrOfCatalogue: data
                    });
                });
        }
    }

    render() {
        let { updateState } = this.props;
        return (
            <Form>
                <Form.Group>
                    <Form.Control
                        type='text'
                        placeholder='text input for search name'
                        onChange={
                            (e) => {
                                updateState(null, {
                                    nameSearch: e.target.value
                                });
                            }
                        }
                    />
                </Form.Group>
                <Form.Group>
                    <Form.Label>PriceMore:</Form.Label>
                    <Form.Control
                        type='number'
                        placeholder='0'
                        min={0}
                        onChange={(e) => {
                            updateState(null, {
                                priceMore: e.target.value
                            });
                        }}
                    />
                </Form.Group>
                <Form.Group>
                    <Form.Label>PriceLess:</Form.Label>
                    <Form.Control
                        type='number'
                        placeholder='0'
                        min={0}
                        onChange={(e) => {
                            updateState(null, {
                                priceLess: e.target.value
                            });
                        }}
                    />
                </Form.Group>
                <Form.Group>
                    <Form.Label>Category:</Form.Label>
                    <Form.Control
                        as='select'
                        defaultValue = 'Select category'
                        onChange={(e) => {
                            updateState(null, {
                                id_catalogue: e.target.value
                            });
                        }}
                    >
                        <option disabled>Select category</option>
                        {this.state.arrOfCatalogue.map((el,i) => {
                            return (
                                <option key={`cat-filter-${i}`} value={el.id_catalogue}>
                                    {`${el.name}`}
                                </option>);
                        })}
                    </Form.Control>
                </Form.Group>
            </Form>

        );
    }
}

export default FilterGoodsPanel;