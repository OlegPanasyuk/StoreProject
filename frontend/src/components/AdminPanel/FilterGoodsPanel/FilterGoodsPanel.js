import React, { Component } from 'react';
import { Form } from 'react-bootstrap';
import PropTypes from 'prop-types';

import CatalogueService from '../../../Services/CatalogueService';

export class FilterGoodsPanel extends Component {
    constructor(props) {
        super(props);
        this.state = {
            arrOfCatalogue: []
        };
    }

    // eslint-disable-next-line camelcase
    UNSAFE_componentWillMount() {
        CatalogueService.getCatalogue()
            .then((data) => {
                this.setState({
                    arrOfCatalogue: data.data
                });
            });
    }

    render() {
        const { updateState } = this.props;
        const { arrOfCatalogue } = this.state;
        return (
            <Form>
                <Form.Group>
                    <Form.Control
                        type='text'
                        placeholder='text input for search name'
                        onChange={(e) => {
                            updateState(null, {
                                nameSearch: e.target.value
                            });
                        }}
                    />
                </Form.Group>
                <Form.Group>
                    <Form.Label>PriceMore:</Form.Label>
                    <Form.Control
                        type='number'
                        placeholder='0'
                        min={0}
                        onChange={(e) => {
                            updateState(null, {
                                priceMore: e.target.value
                            });
                        }}
                    />
                </Form.Group>
                <Form.Group>
                    <Form.Label>PriceLess:</Form.Label>
                    <Form.Control
                        type='number'
                        placeholder='0'
                        min={0}
                        onChange={(e) => {
                            updateState(null, {
                                priceLess: e.target.value
                            });
                        }}
                    />
                </Form.Group>
                <Form.Group>
                    <Form.Label>Category:</Form.Label>
                    <Form.Control
                        as='select'
                        defaultValue='Select category'
                        onChange={(e) => {
                            updateState(null, {
                                id_catalogue: e.target.value
                            });
                        }}
                    >
                        <option disabled>Select category</option>
                        {arrOfCatalogue.map(el => (
                            <option key={`cat-filter-${el.id_catalogue}`} value={el.id_catalogue}>
                                {`${el.name}`}
                            </option>
                        ))}
                    </Form.Control>
                </Form.Group>
            </Form>

        );
    }
}

FilterGoodsPanel.propTypes = {
    updateState: PropTypes.func
};

FilterGoodsPanel.defaultProps = {
    updateState: () => null
};

export default FilterGoodsPanel;
