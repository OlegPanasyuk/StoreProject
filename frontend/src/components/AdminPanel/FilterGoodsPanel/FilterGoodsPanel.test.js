import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import { FilterGoodsPanel } from './FilterGoodsPanel';


configure({ adapter: new Adapter() });

function setupComponent(props = {
    updateState: () => { }
}) {
    const component = shallow(<FilterGoodsPanel {...props} />);
    return { component, props };
}

describe('EditGoodsItem --- Snapshot', () => {
    it('Component must be rendered', () => {
        const { component } = setupComponent();
        expect(FilterGoodsPanel.propTypes.updateState).toBe(PropTypes.func);
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});
