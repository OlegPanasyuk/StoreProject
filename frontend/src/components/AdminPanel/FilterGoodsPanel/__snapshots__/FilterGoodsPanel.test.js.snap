// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditGoodsItem --- Snapshot Component must be rendered 1`] = `
<Form
  as="form"
  inline={false}
>
  <FormGroup
    as="div"
  >
    <FormControl
      as="input"
      onChange={[Function]}
      placeholder="text input for search name"
      type="text"
    />
  </FormGroup>
  <FormGroup
    as="div"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      PriceMore:
    </FormLabel>
    <FormControl
      as="input"
      min={0}
      onChange={[Function]}
      placeholder="0"
      type="number"
    />
  </FormGroup>
  <FormGroup
    as="div"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      PriceLess:
    </FormLabel>
    <FormControl
      as="input"
      min={0}
      onChange={[Function]}
      placeholder="0"
      type="number"
    />
  </FormGroup>
  <FormGroup
    as="div"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Category:
    </FormLabel>
    <FormControl
      as="select"
      defaultValue="Select category"
      onChange={[Function]}
    >
      <option
        disabled={true}
      >
        Select category
      </option>
    </FormControl>
  </FormGroup>
</Form>
`;
